### PR TITLE
Pods 9185

### PR DESCRIPTION
**Pull request by PavelVjalicin**

### CODE REVIEW

File: app/controllers/actions/AuthAction.scala [hunk 1]
```diff
@@ -176,13 +188,16 @@ abstract class AuthenticatedAuthAction @Inject()(
...
 
-  private def handleFailure: PartialFunction[Throwable, Result] = {
+  private def handleFailure(request: RequestHeader): PartialFunction[Throwable, Result] = {
     case _: NoActiveSession =>
       Redirect(config.loginUrl, Map("continue" -> Seq(config.loginContinueUrl)))
     case _: InsufficientEnrolments =>
       Redirect(controllers.routes.UnauthorisedController.onPageLoad())
     case _: InsufficientConfidenceLevel =>
-      Redirect(controllers.routes.UnauthorisedController.onPageLoad())
+      val completionURL = request.uri
```
[the remark this answers is not in the document]
Comment (PavelVjalicin) [reply app/controllers/actions/AuthAction.scala]: This was tested to work.

File: build.sbt [hunk 2]
```diff
@@ -21,7 +21,7 @@ lazy val root = (project in file("."))
     ScoverageKeys.coverageMinimumStmtTotal := 80,
     ScoverageKeys.coverageFailOnMinimum := true,
     ScoverageKeys.coverageHighlighting := true,
-    scalacOptions ++= Seq("-Xfatal-warnings", "-feature"),
+    scalacOptions ++= Seq(/*"-Xfatal-warnings",*/ "-feature"),
```
[the remark this answers is not in the document]
Comment (PavelVjalicin) [reply build.sbt]: Removed entirely.

File: app/config/FrontendAppConfig.scala [hunk 3]
```diff
@@ -93,6 +97,7 @@ class FrontendAppConfig @Inject()(configuration: Configuration, servicesConfig:
   lazy val hmrcTaxHelplineUrl: String = configuration.get[String]("urls.hmrcTaxHelpline")
 
   lazy val loginContinueUrl: String = configuration.get[String]("urls.loginContinue")
+  lazy val loginContinueUrlRelative: String = configuration.get[String]("urls.loginContinueRelative")
```
[the remark this answers is not in the document]
Comment (PavelVjalicin) [reply app/config/FrontendAppConfig.scala:100]: Fixed.

File: app/config/FrontendAppConfig.scala [hunk 4]
```diff
@@ -24,6 +24,8 @@ import play.api.i18n.Lang
 import play.api.mvc.Call
 import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 
+import java.net.URLEncoder
```
[the remark this answers is not in the document]
Comment (PavelVjalicin) [reply app/config/FrontendAppConfig.scala]: Removed

**Review by adamconder — APPROVED** [review]

**Review by vikramsvr — APPROVED** [review]

**Pull request merged**